addons/pos_coupon/models/pos_config.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+# NOTE Use black to automatically format this code.
+
+from datetime import datetime
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class PosConfig(models.Model):
+    _inherit = "pos.config"
+
+    use_coupon_programs = fields.Boolean(
+        "Coupons & Promotions",
+        help="Use coupon and promotion programs in this PoS configuration.",
+    )
+    coupon_program_ids = fields.Many2many(
+        "coupon.program",
+        relation="coupon_program_pos_config_rel",
+        string="Coupon Programs",
+        domain=[("program_type", "=", "coupon_program"), ("active", "=", True)],
+    )
+    promo_program_ids = fields.Many2many(
+        "coupon.program",
+        relation="promo_program_pos_config_rel",
+        string="Promotion Programs",
+        domain=[("program_type", "=", "promotion_program"), ("active", "=", True)],
+    )
+    program_ids = fields.Many2many(
+        "coupon.program",
+        relation="program_pos_config_rel",
+        string="Coupons and Promotions",
+        store=True,
+        compute="_compute_program_ids",
```

Comment:
I don't understand why it is compute and stored?, as it is a many2many field whit a relation table, probably not needed.

if the problem comes from the fact that tere are multiple field on the same table, I'd let this field being the main one, and the two others,computed based on domains.

Reply:
We need the separation because we are introducing form elements in pos config form for both the coupon and promo programs. This is done because even though coupon and promo programs are both 'coupon.program', they are still inherently discriminated. In this arrangement, user can specify Coupon Programs separate from Promotion Programs. We are just combining them in program_ids.

I check the code and there is not much use of program_ids, perhaps we can get away from it? What do you think?

Reply:
promo_program_ids and coupon_program_ids are just here to ease the encoding of the programs, but the modelisation is only one many2many between pos_config and coupon.program.

So for me the correct implementation is this one:
program_ids = fields.Many2many("coupon.program", string="Coupons and Promotions")
promo_program_ids = fields.Many2many("coupon.program", compute="filter_on_domain",inverse="set_values_in_program_ids")
coupon_program_ids = fields.Many2many("coupon.program",compute="filter_on_domain",inverse="set_values_in_program_ids")

Reply:
I see. I wasn't aware of that kind of modelling. I'll try this recommendation.